src/civetweb.c

```diff
@@ -12422,9 +12442,249 @@ consume_socket(struct mg_context *ctx, struct socket *sp)
 #undef QUEUE_SIZE
 }
 
+/* Worker threads take accepted write connection from the queue */
```

Comment:
All the duplicated code can be reduced by setting up wrappers that call an _int() function with appropriate params; the _int() function will have the common code, and the wrapper will set up params for the call.